Benchmark Tests for FNO and DeepONets

ayushinav opened the pull request:
To fix #13 

* [x] comparison with [neuraloperators](https://github.com/neuraloperator/neuraloperator)
* [x] comparison with [deepxde](https://github.com/lululxvi/deepxde)
* [x] comparison with Flux variant
* [ ] GPU comparisons 

I had some issues with CUDA and all when installing torch CUDA toolkit.

yuehhua commented:
This is not a Markov Neural Operators.

foldfelis commented:
> This is not a Markov Neural Operators.

Please take **closer** look at the model introduced by the [paper](https://arxiv.org/abs/2106.06898).

yuehhua commented on README.md:
```diff
@@ -68,25 +68,25 @@ PDE training examples are provided in `example` folder.
 - [x] One-dimensional Burgers' equation example
-- [ ] Two-dimensional Darcy flow equation example
-- [ ] Two-dimensional Navier-Stokes equation example
+- [ ] `Markov Neural Operators`
```
```suggestion
- [ ] `MarkovOperator` layer
```

yuehhua commented on README.md:
```diff
@@ -68,25 +68,25 @@ PDE training examples are provided in `example` folder.
-- [ ] Two-dimensional Darcy flow equation example
-- [ ] Two-dimensional Navier-Stokes equation example
+- [ ] `Markov Neural Operators`
+- [x] Double Pendulum prediction example
```
```suggestion
- [x] Double pendulum prediction example
```

foldfelis commented on example/DoublePendulum/src/data.jl:
```diff
@@ -0,0 +1,54 @@
+    register(DataDep(
+        "DoublePendulumChaotic",
+        """
+        Dataset was generated on the basis of 21 individual runs of a double pendulum.
```
This meta was copy from home page of dataset.

ayushinav committed:
benchmarking with python codes

ayushinav committed:
updated versions

avik-pal committed:
fix: cleanup the benchmarking code

avik-pal committed:
feat: setup basic flux code

[skip ci] [skip docs]

ayushinav committed:
Fair tests across Lux, Flux, and pytorch

ayushinav committed:
minor cleanup

avik-pal committed:
perf: use fast_activation instead of broadcasting

avik-pal committed:
perf: use the permuted formulation

dependabot[bot] committed:
chore: bump crate-ci/typos from 1.24.5 to 1.24.6 (#38)

Bumps [crate-ci/typos](https://github.com/crate-ci/typos) from 1.24.5 to 1.24.6.
- [Release notes](https://github.com/crate-ci/typos/releases)
- [Changelog](https://github.com/crate-ci/typos/blob/master/CHANGELOG.md)
- [Commits](https://github.com/crate-ci/typos/compare/v1.24.5...v1.24.6)

---
updated-dependencies:
- dependency-name: crate-ci/typos
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
